app/services/license_and_rights_statements_setter.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+# Sets the license and rights statements (copyright, use, and reproduction statements)
+class LicenseAndRightsStatementsSetter
+  # @param [Ability] ability ability representing current user and perms
+  # @param [String] druid a druid identifier
+  # @param [String] copyright_statement copyright statement text
+  # @param [String] license license code # TODO: Replace with URI
+  # @param [String] use_statement use statement text
+  # @raise [Dor::Services::Client::UnexpectedResponse] when dor-services-app barfs for some reason
+  # @raise [Dor::Services::Client::NotFoundResponse] when dor-services-app cannot find the given object ID
+  # @raise [RuntimeError] when given identifer does not correspond to an item or collection, when given
+  #                       ability lacks manage_item permission, or when an underlying exception is raised
+  # @return [Cocina::Models::Collection,Cocina::Models::DRO] the cocina object with updates applied
+  def self.set(ability:, druid:, copyright_statement: nil, license: nil, use_statement: nil)
+    new(ability: ability,
+        druid: druid,
+        copyright_statement: copyright_statement,
+        license: license,
+        use_statement: use_statement)
+      .set
+  end
+
+  # @param [Ability] ability ability representing current user and perms
+  # @param [String] druid a druid identifier
+  # @param [String] copyright_statement copyright statement text
+  # @param [String] license license code # TODO: Replace with URI
+  # @param [String] use_statement use statement text
+  def initialize(ability:, druid:, copyright_statement: nil, license: nil, use_statement: nil)
+    @ability = ability
+    @druid = druid
+    @copyright_statement = copyright_statement
+    @license = license
+    @use_statement = use_statement
+  end
+
+  # @raise [Dor::Services::Client::UnexpectedResponse] when dor-services-app barfs for some reason
+  # @raise [Dor::Services::Client::NotFoundResponse] when dor-services-app cannot find the given object ID
+  # @raise [RuntimeError] when given identifer does not correspond to an item or collection, when given
+  #                       ability lacks manage_item permission, or when an underlying exception is raised
+  # @return [Cocina::Models::Collection,Cocina::Models::DRO] the cocina object with updates applied
+  def set
+    raise "#{druid} is not an item or collection (#{cocina_object.type})" unless cocina_object.dro? || cocina_object.collection?
```

Comment:
i'm ok with this as is;  should/could it be raise if it's an APO?   not sure if we have other categories

Reply:
yeah, this'll raise if passed an APO. there are a couple other things we could conceivably get here.